src/main/java/net/dv8tion/jda/core/managers/impl/AudioManagerImpl.java

```diff
@@ -65,8 +66,8 @@
 
     public AudioManagerImpl(Guild guild)
     {
-        this.guild = guild;
-        this.api = (JDAImpl) guild.getJDA();
+        this.guild = (GuildImpl) guild;
```

Comment:
Correct me if I'm wrong, but doesn't this defy the purpose of Guild being an interface at all?
If it's an interface, one can write their own implementation of it, however, here only GuildImpl works.
(I'm not too sure, so please enlighten me.)